### PR TITLE
Add support for prefix unary plus operator

### LINKED ISSUE #120: Support prefix unary '+'?

### PR DESCRIPTION
**Pull request by irevoire**
Fix #120

Hey, I was looking for easy-to-implement issues just to improve my understanding of the code base.
Let me know if you have other ideas

### CODE REVIEW

File: numbat-cli/tests/integration.rs [hunk 1]
```diff
@@ -33,8 +33,8 @@ fn pass_expression_on_command_line() {
         .arg("--expression")
         .arg("2 ++ 3")
         .assert()
-        .failure()
-        .stderr(predicates::str::contains("while parsing"));
+        .success()
+        .stdout(predicates::str::contains("5"));
```
Comment (sharkdp) [inline numbat-cli/tests/integration.rs:37]: This test was here to make sure that we fail with a non-zero exit status on parser errors. I suggest we revert this and change the expression instead (e.g. `2 /+ 3`.

**Review by sharkdp — APPROVED** [review]
Thank you very much! Just one minor comment.

File: numbat-cli/tests/integration.rs (hunk 1 above)
Comment (irevoire) [reply numbat-cli/tests/integration.rs:37]: Yeah, except, hehe, now `2 / +3` is a valid expression 😎 

I updated it to `2 +/ 3`

Comment (sharkdp) [reply numbat-cli/tests/integration.rs]: Oh. I should have invested more than 2 seconds of thought into that example :smile:

**Review by sharkdp — APPROVED** [review]
Thank you

**Pull request merged**